File: src/main/java/de/iisys/drossner/algodat/sprachbeispiel/Beispiel.java
package de.iisys.drossner.algodat.sprachbeispiel;

import java.util.stream.IntStream;

/**
 * created at 11.01.18
 *
 * @author drossner
 */
public class Beispiel {
    public static void main(String[] args){
        final int n = 3;
        int ergebnis = 0;
        for(int i = n; i > 0; i--){
            ergebnis += i;
        }
        System.out.println(ergebnis);
    }

    //für Fortgeschrittene:
    public static void kuerzer(){
        final int n = 3;
        System.out.println( IntStream.range(n, 0).sum() );
    }
}

File: src/main/kotlin/de/iisys/drossner/algodat/sprachbeispiel/Beispiel.kt
package de.iisys.drossner.algodat.sprachbeispiel

/**
 * created at 11.01.18
 * @author drossner
 */
//berechne n + n-1 + n-2 + ... + 1
fun main(args: Array<String>){
    val n = 3
    var ergebnis = 0
    for(i in n downTo  1){
        ergebnis += i
    }
    println(ergebnis)
}

//für Fortgeschrittene:
fun kuerzer(){
    val n = 3
    println( (n downTo 1).sum() )
}